webapi/src/main/scala/org/knora/webapi/routing/Authenticator.scala

```diff
@@ -462,39 +600,117 @@ object Authenticator {
+    private def getUserProfileV1ByEmail(email: String)(implicit system: ActorSystem, timeout: Timeout, executionContext: ExecutionContext): UserProfileV1 = {
+
         val responderManager = system.actorSelection(RESPONDER_MANAGER_ACTOR_PATH)
 
         if (email.nonEmpty) {
-            // try to get it from the cache
-            CacheUtil.get[UserProfileV1](AUTHENTICATION_CACHE_NAME, email) match {
-                case Some(userProfile) =>
-                    // found a user profile in the cache
-                    log.debug(s"getUserProfileByEmail - cache hit: $userProfile")
-                    userProfile
-                case None =>
-                    // didn't found one, so I will try to get it from the triple store
-                    val userProfileV1Future = for {
-                        maybeUserProfileV1 <- (responderManager ? UserProfileByEmailGetV1(email, UserProfileType.FULL)).mapTo[Option[UserProfileV1]]
-                        userProfileV1 = maybeUserProfileV1 match {
-                            case Some(up) => up
-                            case None => {
-                                log.debug(s"getUserProfileByEmail - supplied email not found - throwing exception")
-                                throw BadCredentialsException(s"$BAD_CRED_USER_NOT_FOUND")
-                            }
-                        }
-                        _ = CacheUtil.put(AUTHENTICATION_CACHE_NAME, email, userProfileV1)
-                        _ = log.debug(s"getUserProfileByEmail - from triplestore: $userProfileV1")
-                    } yield userProfileV1
-
-                    // TODO: return the future here instead of using Await.
-					Await.result(userProfileV1Future, Duration(3, SECONDS))
-            }
+            val userProfileV1Future = for {
+                maybeUserProfileV1 <- (responderManager ? UserProfileByEmailGetV1(email, UserProfileTypeV1.FULL)).mapTo[Option[UserProfileV1]]
+                userProfileV1 = maybeUserProfileV1 match {
+                    case Some(up) => up
+                    case None => {
+                        log.debug(s"getUserProfileV1ByEmail - supplied email not found - throwing exception")
+                        throw BadCredentialsException(s"$BAD_CRED_USER_NOT_FOUND")
+                    }
+                }
+                _ = log.debug(s"getUserProfileV1ByEmail - userProfileV1: $userProfileV1")
+            } yield userProfileV1
+
+            // TODO: return the future here instead of using Await.
+            Await.result(userProfileV1Future, Duration(3, SECONDS))
+        } else {
+            throw BadCredentialsException(BAD_CRED_EMAIL_NOT_SUPPLIED)
+        }
+    }
+}
+
+object JWTHelper {
+
+    import Authenticator.AUTHENTICATION_INVALIDATION_CACHE_NAME
+
+    val algorithm = Algorithm.HS256
+    val requiredHeaders: Set[HeaderField] = Set[HeaderField](Typ)
+    val requiredClaims: Set[ClaimField] = Set[ClaimField](Iss, Sub, Aud, Iat, Exp, Jti)
```

Comment:
Please add comments explaining what is being configured here.

Reply:
done